### PR TITLE
Improve diagnostics for import failures

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://76217088

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -2868,52 +2868,51 @@ class StoringDiagnosticConsumer : public swift::DiagnosticConsumer {
...
-                  diagnostic.line - first_line + 1);
-              start_pos = match_pos + match_len;
-              match_pos = diagnostic.description.find(match.GetString().str(),
-                                                      start_pos);
-            } while (match_pos != std::string::npos);
-
-            // Append any last remaining text.
-            if (start_pos < diagnostic.description.size())
+      if (first_line > 0 && bufferID != UINT32_MAX) {
+        // Make sure the error line is in range or in another file.
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:2872]: > or in another file

maybe I don't know this code, but is this supposed to be "and not in another file"

> diagnostic.bufferID == bufferID

**Review by kastiglione — APPROVED** [review]

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:2872]: The code wants to suppress all errors that are in the main buffer and out of range. It displays all errors that are either
- in the main buffer inside the range
- in another file

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Pull request merged**